### DISCUSSION TITLE
Windows

### DISCUSSION THREAD

**Discussion by gamemaster7720** [dawigit/picoclock#8] [Q&A]
hey i just got myself a RP2040-LCD-1.28 and tried to use your code but i was wondering how would i make a uf2 bc everytime i try to use cd picoclock;mkdir build;cd build;cmake ..;make

i get 

PICO_SDK_PATH is C:/Users/Default/Downloads/Compressed/pico-sdk-master_2/pico-sdk-master
PICO platform is rp2040.
Build type is Debug
Using regular optimized debug build (set PICO_DEOPTIMIZED_DEBUG=1 to de-optimize)
Using board configuration from C:/Users/Default/Downloads/Compressed/pico-sdk-master_2/pico-sdk-master/src/boards/include/boards/pico.h
CMake Warning at C:/Users/Default/Downloads/Compressed/pico-sdk-master_2/pico-sdk-master/src/rp2_common/tinyusb/CMakeLists.txt:10 (message):
  TinyUSB submodule has not been initialized; USB support will be unavailable

  hint: try 'git submodule update --init' from your SDK directory
  (C:/Users/Default/Downloads/Compressed/pico-sdk-master_2/pico-sdk-master).

-- Configuring done (1.0s)
-- Generating done (0.7s)
-- Build files have been written to: C:/Users/Default/Downloads/Compressed/picoclock-main_2/picoclock-main/build
make: *** No targets specified and no makefile found.  Stop.

and for some reasson i just can't get it to work on windows and i was curious if you might had a step by step exaplanation on how i can make this work ? or what i need to install on linux to get it to work

**Comment by dawigit** [comment]
_**hint: try 'git submodule update --init' from your SDK directory**_

You have to install the complete SDK. [9.2. Building on MS Windows](https://datasheets.raspberrypi.com/pico/getting-started-with-pico.pdf)

**Reply by gamemaster7720** [reply]
oke and would you maybe also know how i could fix this error this is linux btw. it crashes on 95%

/usr/lib/gcc/arm-none-eabi/12.2.1/../../../arm-none-eabi/bin/ld: CMakeFiles/main.dir/main.c.obj: in function `get_acid':
/home/matti/Downloads/picoclock-main/main.c:1189: undefined reference to `get_ac'
/usr/lib/gcc/arm-none-eabi/12.2.1/../../../arm-none-eabi/bin/ld: CMakeFiles/main.dir/main.c.obj: in function `print_font_table4':
/home/matti/Downloads/picoclock-main/main.c:1242: undefined reference to `get_ac'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/main.dir/build.make:1321: main.elf] Error 1
make[1]: *** [CMakeFiles/Makefile2:1547: CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

**Reply by dawigit** [reply]
does it work now? get_ac() should be known. if not 4WARD it.

**Comment by gamemaster7720** [comment]
and how would i change the background or theme?

**Reply by dawigit** [reply]
With an attached button OR with minicom/pigsh.

**Reply by gamemaster7720** [reply]
could i change it as a custom background or is that to hard?

**Reply by dawigit** [reply]
Yes, sure! It is meant to be customized.
That's why the scripts are there.
A spinning background is more difficult.
Read tools/tools.md 
If don't come along check back - we can do a step-by-step (guide)

**Reply by gamemaster7720** [reply]
I really tried but i can just not get it to work or change anything

**Reply by dawigit** [reply]
So are you able to build the image? Is 'make' working (without errors)?

**Reply by gamemaster7720** [reply]
Yes that totally works now

**Reply by dawigit** [reply]
Very good!

## Now the step-by-step guide:
- install imagemagick (sudo apt install imagemagick)
- download https://en.wikipedia.org/wiki/Homer_Simpson#/media/File:Homer_Simpson_2006.png
- the png image is 190x390 pixel in size -> our goal is 256x256 in size
`cd ~/picoclock/img`
`cp ~/Downloads/Homer_Simpson_2006.png hosi2006.png`

### now we convert the image from 'alpha-mode' (checker-board background when viewed with eog) to non-alpha:

`convert hosi2006.png -transparent magenta -background black -alpha background -alpha off hosi2006a.png`

### next step is changing the size so it is quadratic

`convert hosi2006a.png -extent 390x390 -roll +88 hosi.png`

#### the '-roll +88' rolls Homer in the center

### now we shrink J to a size of 256x256

`convert hosi.png -resize 256x256 hosi256.png'

### and finally convert png to raw (img2data script in picoclock/tool/)

`img2data hosi256.png 256*256`

This gives us 'hosi256.h' , which we now have to include in the source.
We will now substitute some image in the source with our new hosi256 background.

So in 'main.c' search for 'bg4_256.h'.

The first location should be at line 56:
`#include "img/bg4_256.h"`

change it to:
`#include "img/hosi256.h"`

we continue searching with 'bg4_256' (WITHOUT .h)
and should now be at line 653:

`bg1_256,bg2_256,bg3_256,bg4_256`

change it to
`bg1_256,bg2_256,bg3_256,hosi256`

save!

### done

now build (not in DEVMODE, there are only 3 backgrounds in DEVMODE)
and flash the image. 
in pigsh: 
`bg 8`

**Reply by gamemaster7720** [reply]
Yes i will try that and how could i change the clock pointers? Because i would like to remove the flags
